### ISSUE TITLE
TextHelper autoLink is including newlines in href

### ISSUE THREAD

**Issue by fullybaked** [cakephp/cakephp#7976]
We're using the following code to output some content that is stored in plaintext

```
<?= $this->Text->autoParagraph($this->Text->autoLink($project->summary));?>
```

This is causing links in the summary to render as 

```
<a href="http://www.example.com</br>">www.example.com</a>
<a href="http://www.example.com</br>">www.example.com</a>
```

if the plaintext stored is

```
www.example.com
www.example.com
```

I wonder if the `$url` here https://github.com/cakephp/cakephp/blob/master/src/View/Helper/TextHelper.php#L167 needs a trim to ensure no whitespace creeps in (it's not valid anyway).

I don't mind putting in a PR, it would only take a second, but would like opinion first.  NB, the email equivalent probably needs this too, but it's not part of my use case so I haven't checked.

**Comment by dereuromark** [comment]
Makes sense. Can you make a PR with the suggested change and a test case that proofs it was invalid before and is now fixed?

**Comment by markstory** [comment]
A pull request would be great. The current output is very wrong :smile:

**Comment by fullybaked** [comment]
No problems.  Strangely enough I can't seem to get a test case that fails on the current code, even though I'm seeing the effects in a running app.  I'll dig in and PR as soon as I have all the parts together.

**Comment by dereuromark** [comment]
What OS are u using? I suspect line Ending types

**Comment by fullybaked** [comment]
Ah, good point.  I'm on OSX.  I'll try it with a few different line endings.

**Comment by fullybaked** [comment]
@dereuromark Yes. It is line endings.  

I exported our live data into a text file and checked, and the text that's causing this bug has `0xD 0xA` (CRLF) as the line endings.  It appears as though this makes a line break remain in the href and causes the bug.  But... I can't for the life of me get this into a test case.  Anything I try passes without any fix required, even though I can see the bug in the real world use... :confounded:  

We're using the `trim` solution I suggested in an overridden TextHelper in our src/ directory and it's working fine, but I haven't made the PR without the tests.  

Any tips on testing different line endings.  I've tried strings with `\n` `\r\n` `\r` in them, but they all pass... Help would be much appreciated at this point.
